### PR TITLE
feat: add no-type-alias rule

### PR DESCRIPTION
**Pull request by ScriptedAlchemy**
## Summary
- add the `@typescript-eslint/no-type-alias` rule implementation and docs
- register the rule in the TypeScript plugin registry
- expand Go coverage for options and edge cases
- enable the rstest fixture and generated snapshots

## Validation
- `act` Linux `test-go` job passed: `go test -parallel 8 ./internal/...`
- `act` Linux `lint` job passed: `golangci-lint`, `npm run lint:go`, `npm run format:go`, `pnpm check-spell`
- `act` Linux `test-node` job passed with pnpm store isolated outside the workspace (`PNPM_HOME=/tmp/pnpm-home`, `npm_config_store_dir=/tmp/pnpm-store`)
  - `packages/rslint-test-tools`: 383 test files passed, 1155 passed / 30 skipped
  - VSCode extension tests passed
- `act` Linux `test-wasm` job passed; wasm build and JS wrapper completed

## Notes
- macOS and Windows jobs were intentionally not run locally; Docker/act cannot faithfully emulate those runners.
- The branch was prepared directly on upstream `web-infra-dev/rslint`, not the fork.

### CODE REVIEW

File: internal/plugins/typescript/rules/no_type_alias/no_type_alias.go [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+	if composition == compositionUnion {
+		compositionName = "union"
+	}
+
+	return rule.RuleMessage{
+		Id:          "noCompositionAlias",
+		Description: typeName + " in " + compositionName + " types are not allowed.",
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (ScriptedAlchemy) [reply internal/plugins/typescript/rules/no_type_alias/no_type_alias.go:122]: Addressed in `b4f9d602`: added the `unwrapParenthesized` helper at line 123.

File: internal/plugins/typescript/rules/no_type_alias/no_type_alias.go [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+				ast.KindStringKeyword,
+				ast.KindSymbolKeyword,
+				ast.KindUndefinedKeyword,
+				ast.KindUnknownKeyword,
+				ast.KindVoidKeyword:
+				return true
+			default:
+				return false
+			}
+		}
```
[the remark this answers is not in the document]
Comment (ScriptedAlchemy) [reply internal/plugins/typescript/rules/no_type_alias/no_type_alias.go:247]: Addressed in `b4f9d602`: `isAliasKind` and `isKeywordType` now unwrap parenthesized types before checking the kind.

File: internal/plugins/typescript/rules/no_type_alias/no_type_alias.go [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+			return typeOp.Type.Kind == ast.KindTupleType
+		}
+
+		isValidGeneric := func(t typeWithComposition) bool {
+			if t.node == nil || t.node.Kind != ast.KindTypeReference {
+				return false
+			}
+			typeRef := t.node.AsTypeReferenceNode()
+			return typeRef != nil && typeRef.TypeArguments != nil && len(typeRef.TypeArguments.Nodes) > 0
+		}
```
[the remark this answers is not in the document]
Comment (ScriptedAlchemy) [reply internal/plugins/typescript/rules/no_type_alias/no_type_alias.go:278]: Addressed in `b4f9d602`: `isValidTupleType` and `isValidGeneric` now unwrap parenthesized types before classification.

File: internal/plugins/typescript/rules/no_type_alias/no_type_alias.go [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+					}
+					if typeOp.Operator == ast.KindReadonlyKeyword && typeOp.Type != nil && isAliasKind(typeOp.Type) {
+						checkAndReport(opts.AllowAliases, isTopLevel, t, "Aliases")
+						return
+					}
+				}
+			}
+
+			reportError(t.node, t.composition, isTopLevel, "Unhandled")
+		}
```
[the remark this answers is not in the document]
Comment (ScriptedAlchemy) [reply internal/plugins/typescript/rules/no_type_alias/no_type_alias.go:353]: Addressed in `b4f9d602`: `validateTypeAlias` now unwraps parenthesized types and handles the `unique symbol` type operator correctly.

File: internal/plugins/typescript/rules/no_type_alias/no_type_alias.go [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+
+			if isKeywordType(node) || isAliasKind(node) {
+				checkAndReport(opts.AllowAliases, isTopLevel, t, "Aliases")
+				return
+			}
+
+			if node.Kind == ast.KindTypeOperator {
+				typeOp := node.AsTypeOperatorNode()
+				if typeOp != nil {
+					if typeOp.Operator == ast.KindKeyOfKeyword || typeOp.Operator == ast.KindUniqueKeyword {
```
Comment (fansenze) [inline internal/plugins/typescript/rules/no_type_alias/no_type_alias.go]: **`unique symbol` is misclassified as an alias — behavioral divergence from typescript-eslint**

This line treats the `unique` type operator as an alias. Upstream's `no-type-alias` only treats `keyof` and `readonly` (over an alias operand) as aliases — `unique symbol` falls through to the final `else` and is reported as `'Unhandled'`, which **no `allow*` option suppresses**.

I verified both implementations by running them directly (upstream `@typescript-eslint/eslint-plugin@7.18.0` via the `Linter` API; this PR's binary via `@rslint/core`'s `lint()`):

| input | upstream v7.18.0 | this PR |
|---|---|---|
| `type Foo = unique symbol` (default) | `Type unhandled are not allowed.` | `Type aliases are not allowed.` |
| `type Foo = unique symbol` (`allowAliases: 'always'`) | **still reports** | **VALID (not reported)** |
| `type Foo = keyof [string] \| unique symbol` | `[1:29] Unhandled in union types are not allowed.` | `[1:29] Aliases in union types are not allowed.` |

The `allowAliases: 'always'` row is a real **false negative** — code upstream flags is silently accepted here. It also matches upstream's own golden fixture (`typeName: 'Unhandled'` at col 29), kept verbatim in this PR's `no-type-alias.test.ts`; it only passes because `checkDiagnosticEqual` compares `messageId` + line/column, not `data`/message text.

Suggested fix — drop `unique` so it reaches the `"Unhandled"` branch at line 352:
```go
if typeOp.Operator == ast.KindKeyOfKeyword {
```

Note: this will (correctly) make `type Foo = unique symbol` with `allowAliases: 'always'` report an error, so the matching valid test case in `no_type_alias_test.go` (line 22) should move to `invalid` (`noTypeAlias`, line 1, column 12).

Comment (ScriptedAlchemy) [reply internal/plugins/typescript/rules/no_type_alias/no_type_alias.go]: Addressed in 3f15df53: unique symbol no longer goes through the allowAliases path, the Go regression moved the allowAliases: always case to invalid, and the rstest fixture/snapshot now expect the Unhandled diagnostic.

**Review by fansenze — APPROVED** [review]
